.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,106 @@
+# SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Install and Run tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  linter:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11" ]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: cache poetry install
+        uses: actions/cache@v3
+        with:
+          path: ~/.local
+          key: poetry-${{matrix.os}}-1.4.0-0
+
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.4.0
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      - name: cache deps
+        id: cache-deps
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: pydeps-${{ matrix.os}}-${{matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install package dependencies
+        run: poetry install --no-interaction --no-root
+        if: steps.cache--deps.outputs.cache-hit != 'true'
+
+      - run: poetry install --no-interaction
```

Comment:
To be honest, I don't completely understand this here: 
Somehow we run `poetry install` if there is no cache (line 48-50) and then every time once again (line 52)?
Might be fine though, I don't know much about Github action syntax.

Reply:
I got the workflow from this [exampe](https://jacobian.org/til/github-actions-poetry/). The idea is to use a cache for the dependencies but not for the package itself. Should I add a comment in the workflow to clarify this?

Reply:
Ah, OK.
Yes, maybe it is worth a comment.